proxy/proxy.go

```diff
@@ -1456,10 +1456,6 @@ func (p *Proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 			logging.LogAccess(entry, additionalData)
 		}
-		// This flush is required in I/O error
-		if ctx.successfulUpgrade {
-			lw.Flush()
```

Comment:
then we need to make sure we have a Flush in the error path, I think in line https://github.com/zalando/skipper/blob/master/proxy/proxy.go#L847

Reply:
Hi. The full story goes like this: we needed to flush error response and [it was introduced](https://github.com/zalando/skipper/pull/1560/files). Then it was found that [upgrade connections suffer](https://github.com/zalando/skipper/issues/1564) and [this fix](https://github.com/zalando/skipper/pull/1565/files) was done. Apparently the last fix passed review in rush because it simply needs to be negated to work
```go
		if !ctx.successfulUpgrade {
			lw.Flush()
		}
```

Reply:
@szuecs @AlexanderYastrebov, is this the correct fix for this problem?